contracts/0.4.24/nos/NodeOperatorsRegistry.sol

```diff
@@ -1,702 +1,1169 @@
+        _requireValidRange(_offset.add(_limit) <= signingKeysStats.get(TOTAL_KEYS_COUNT_OFFSET));
+
+        uint256 depositedSigningKeysCount = signingKeysStats.get(DEPOSITED_KEYS_COUNT_OFFSET);
+        (pubkeys, signatures) = SigningKeys.initKeySig(_limit);
+        used = new bool[](_limit);
+
+        for (uint256 i; i < _limit; ++i) {
+            SIGNING_KEYS_MAPPING_NAME.loadKeySigAndAppend(_nodeOperatorId, _offset + i, i, pubkeys, signatures);
+            used[i] = (_offset + i) < depositedSigningKeysCount;
+        }
+    }
+
+    /// @notice Returns the type of the staking module
+    function getType() external view returns (bytes32) {
+        return TYPE_POSITION.getStorageBytes32();
+    }
 
-        uint256 k1;
-        uint256 k2;
-        assembly {
-            k1 := mload(add(_key, 0x20))
-            k2 := mload(add(_key, 0x40))
+    function getStakingModuleSummary() external view returns (
+        uint256 totalExitedValidators,
+        uint256 totalDepositedValidators,
+        uint256 depositableValidatorsCount
+    ) {
+        uint256 nodeOperatorsCount = getNodeOperatorsCount();
+
+        uint256 tmpTotalExitedValidators;
+        uint256 tmpTotalDepositedValidators;
+        uint256 tmpDepositableValidatorsCount;
+        for (uint256 i; i < nodeOperatorsCount; ++i) {
+            (tmpTotalExitedValidators, tmpTotalDepositedValidators, tmpDepositableValidatorsCount) =
+                _getNodeOperatorValidatorsSummary(i);
+            totalExitedValidators += tmpTotalExitedValidators;
+            totalDepositedValidators += tmpTotalDepositedValidators;
+            depositableValidatorsCount += tmpDepositableValidatorsCount;
         }
+    }
 
-        return 0 == k1 && 0 == (k2 >> ((2 * 32 - PUBKEY_LENGTH) * 8));
+    function getNodeOperatorSummary(uint256 _nodeOperatorId) external view returns (
+        bool isTargetLimitActive,
+        uint256 targetValidatorsCount,
+        uint256 stuckValidatorsCount,
+        uint256 refundedValidatorsCount,
+        uint256 stuckPenaltyEndTimestamp,
+        uint256 totalExitedValidators,
+        uint256 totalDepositedValidators,
+        uint256 depositableValidatorsCount
+    ) {
+        _onlyExistedNodeOperator(_nodeOperatorId);
+
+        Packed64x4.Packed memory operatorTargetStats = _loadOperatorTargetValidatorsStats(_nodeOperatorId);
+        Packed64x4.Packed memory stuckPenaltyStats = _loadOperatorStuckPenaltyStats(_nodeOperatorId);
+
+        isTargetLimitActive = operatorTargetStats.get(IS_TARGET_LIMIT_ACTIVE_OFFSET) != 0;
+        targetValidatorsCount = operatorTargetStats.get(TARGET_VALIDATORS_COUNT_OFFSET);
+        stuckValidatorsCount = stuckPenaltyStats.get(STUCK_VALIDATORS_COUNT_OFFSET);
+        refundedValidatorsCount = stuckPenaltyStats.get(REFUNDED_VALIDATORS_COUNT_OFFSET);
+        stuckPenaltyEndTimestamp = stuckPenaltyStats.get(STUCK_PENALTY_END_TIMESTAMP_OFFSET);
```

Comment:
If we report 1->0 stuck keys in the first report, and 0->1 in the next one, after a few days, we will have the `stuckPenaltyEndTimestamp` that point to the past. The member will still be penalized. All calculations will work correctly, because isOperatorPenalized takes this situation into account, but here it doesn't seem neat.

Perhaps we should return MAX_UINT here in such a scenario? or even set it explicitly when reporting stuck keys

Reply:
I think this is a standard situation, so we don't need to return max_uint